rango-exchange/rango-client
# fix: improve `TON` signer and `MyTonWallet` provider

# Summary

- Fix the issue where `TON` tokens are not being displayed correctly in the token list
- Return the `non-bounceable` address for the `MytonWallet` provider
- Retrieve and return the hash of the message from the `BOC` after signing the `TON` transaction
- Add an option to retrieve the TON Connect manifest from the widget configuration.

# How did you test this change?

- Review the list of `TON` tokens in the token list. Ensure that if a token has an address or name, it is displayed correctly. You can verify this by checking the meta response from the network tab and comparing it with the UI.
- Connect the `MytonWallet` provider. The address displayed in the confirmation wallet modal must match the address selected and shown in the `MytonWallet` UI.
- Create a transaction using the `TON` blockchain as the source. Sign the transaction and check the network tab to ensure that the hash of the message is sent to the server with the `checkStatus` request.

# Note

There are some considerations for using the provider API instead of the `tonconnect/sdk`. Since the main pull request was made over a year ago, I might not recall everything, but here are a few points:

- I was concerned about creating multiple instances of the TON Connect SDK for each TON Connect-compatible provider we integrated.
- The `tonconnect/sdk` uses callbacks to handle wallet connection events, which added extra effort for implementation.
- Using the SDK will introduce an additional dependency and increase the bundle size, with relatively little benefit in return.

# Checklist:
